convert2rhel/actions/report.py

```diff
@@ -205,6 +206,8 @@ def summary(results, include_all_reports=False, with_colors=True):
                 "variables": message["variables"],
             }
 
+    report_to_txt(combined_results_and_message)
+
```

Comment:
I think we should call this function in the `finally` block inside main (https://github.com/hosekadam/convert2rhel/blob/c6715e0d267e4ccb4f6156d0df206c65e8c1ddb5/convert2rhel/main.py#L204-L206) instead of here. 

I think that the idea of having this type of fucntion is to serve as a "plugin" (additional formats output to the summary). 

To my knowledge, this function is no different from the `summary_as_json` in report.py module.

Reply:
The reason why I'm calling it there is to reuse a maximum of already generated data. But it makes sense to have it in the same place as the `summary_as_json`. I'll prepare some solution for this.

Reply:
It makes total sense! But the data that `summary` receives is the raw data after the action framework execution is done... So, in theory, is correct that you have the `report_to_txt` here, but I think that the idea behind this function when we first did it was to output the general summary to the user, and not use it as a "entrypoint" for other format "plugins" (As far as I remember, of course). 

Moving it to be next to `summary_as_json` will give the sense that this is something external, something that we are plugging to the output and will generate whatever format we tell the function to output. I would envision that the `summary` function would keep clean and have its own way to output the summary to the user without interference from external sources. 
-----

@abadger, I think we have spoken about this before, but do you think that there is room for the action framework to have a module output system? Something that we can do once and append/add new formats when we need/want. 

Currently, we only have: 
1. the summary that goes to the user
2. summary as json
3. summary as text (for satellite)
4. ???

I don't know that if we will ever have a new format that we need to output to someone else, but if that would be the case, maybe we should think about a strategy to plug those different formats into some sort of system. What do you think?

Reply:
@r0x0d I think that makes perfect sense.  It probably doesn't have to be very elaborate.

When it was just summary as json and summary that goes to the user, it was awkward to design a common API because the user summary is driven through logging.  I think you'll have an easy time creating a good API that  summary_as_json and summary_as_text will share and then you'll need to decide whether to plug summary that goes to user into that same framework or keep it separate.